crates/oxc_allocator/src/allocator.rs

```diff
@@ -571,6 +597,12 @@ impl Allocator {
         bytes
     }
 
+    /// Returns the total number of allocations that have been made in this [`Allocator`] instance.
+    #[cfg(feature = "track_allocations")]
+    pub fn num_alloc(&self) -> usize {
+        NUM_ALLOC.load(Relaxed)
```

Comment:
I've never entirely got my head around atomic ordering, but I think `Relaxed` here may not be sufficient. You want to ensure all the increments have been committed before reading the total. For the same reason, I'm not sure if `Relaxed` where the increments occurs needs to be something stronger.

Or maybe sleeping for a short time before loading the total count would work, to allow everything to "catch up".